hashdb.py

```diff
@@ -875,22 +932,24 @@ def hash_scan():
                                                   xor_value=HASHDB_XOR_VALUE,
                                                   algorithms=[])
         if settings_results:
-            idaapi.msg("HashDB configured successfully!\nHASHDB_API_URL: %s\nHASHDB_USE_XOR: %s\nHASHDB_XOR_VALUE: %s\nHASHDB_ALGORITHM: %s\n" % 
-                       (HASHDB_API_URL, HASHDB_USE_XOR, hex(HASHDB_XOR_VALUE), HASHDB_ALGORITHM))
+            idaapi.msg("HashDB configured successfully!\nHASHDB_API_URL: %s\nHASHDB_USE_XOR: %s\nHASHDB_XOR_VALUE: %s\nHASHDB_ALGORITHM: %s\nHASHDB_ALGORITHM_SIZE: %s\n" % 
+                       (HASHDB_API_URL, HASHDB_USE_XOR, hex(HASHDB_XOR_VALUE), HASHDB_ALGORITHM, HASHDB_ALGORITHM_SIZE))
         else:
             idaapi.msg("HashDB configuration cancelled!\n")
             return 
     try:
         # Open waiting window
         ida_kernwin.show_wait_box("HIDECANCEL\nPlease wait...")
-        # Hack to determine if we should use QWORDs or DWORDs
-        is_32bit = True 
-        int_size = 4
-        cpu_info = idaapi.get_inf_structure()
-        if cpu_info.is_64bit():
+        # Determine if we should scan for DWORDs or QWORDs (algorithm size)
+        if HASHDB_ALGORITHM_SIZE == 32:
+            is_32bit = True
+        elif HASHDB_ALGORITHM_SIZE == 64:
             is_32bit = False
-            int_size = 8
-        # Loop through selected range and look up each DWORD
+        else:
+            idaapi.msg("ERROR: Unexpected algorithm size provided.\n")
+            return
+        int_size = HASHDB_ALGORITHM_SIZE / 8 # Assume 8 bits = 1 byte (should be true for all platforms that IDA works on)
```

Comment:
This creates an implied float which turns `ea` into a float when added to it... cast as an int.

Reply:
Will do, didn't know about this 👍